packages/api/src/OsdkMetadata.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface OsdkMetadata {
+  extraUserAgent: string;
```

Comment:
As discussed, this is for the transition of @osdk/client to `client(stack, ontologyRid, tokenProvider)` so that later when you `client(FooObjDef)` that we can get the generator data added to the user agent.